update.check/src/main/java/org/kaazing/gateway/update/check/UpdateCheckListener.java

```diff
@@ -0,0 +1,29 @@
+/**
+ * Copyright 2007-2016, Kaazing Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kaazing.gateway.update.check;
+
+public interface UpdateCheckListener {
+
+    /**
+     * Notifies the listener that a new @GatewayVersion is available
+     * Listeners may be notified more than once about the same update
+     *
+     * @param currentVersion
+     * @param latestGatewayVersion
+     */
+    void newVersionAvailable(GatewayVersion currentVersion, GatewayVersion latestGatewayVersion);
+
```

Comment:
If HQ agrees that `setUpdateCheckService` is not needed anymore here, it would be best to annotate this interface with the @FunctionalInterface annotation.